tests/Feature/StartTest.php

```diff
@@ -23,46 +23,80 @@ public function testStart()
 
     public function testStartOldVersionAllDependencies()
     {
-        resolve(CommandExecutor::class)
-            ->shouldReceive('getOutputBuffer')
-            ->andReturn('0.0.0');
+        $this->mockChecker(
+            '0.0.0',
+            '0.0.0',
+            '0.0.0'
+        );
 
         $this->artisan('start')->assertExitCode(1);
     }
 
     public function testStartOldVersionDockerDependency()
     {
-        resolve(CommandExecutor::class)
-            ->shouldReceive('getOutputBuffer')
-            ->andReturn('0.0.0', CheckDockerComposeVersion::DOCKER_COMPOSE_MIN_VERSION);
+        $this->mockChecker(
+            '0.0.0',
+            Checker::DOCKER_API_MIN_VERSION,
+            Checker::DOCKER_COMPOSE_MIN_VERSION
+        );
 
         $this->artisan('start')->assertExitCode(1);
     }
 
-    public function testStartOldVersionDockerComposeDependency()
+    public function testStartOldVersionDockerAPIDependency()
     {
-        resolve(CommandExecutor::class)
-            ->shouldReceive('getOutputBuffer')
-            ->andReturn(CheckDockerVersion::DOCKER_MIN_VERSION, '0.0.0');
+        $this->mockChecker(
+            Checker::DOCKER_MIN_VERSION,
+            '0.0.0',
+            Checker::DOCKER_COMPOSE_MIN_VERSION
+        );
 
         $this->artisan('start')->assertExitCode(1);
     }
 
-    public function testStartTimeout()
+    public function testStartOldVersionDockerComposeDependency()
     {
-        resolve(CommandExecutor::class)
-            ->shouldReceive('getOutputBuffer')
-            ->andReturn(CheckDockerVersion::DOCKER_MIN_VERSION, CheckDockerComposeVersion::DOCKER_COMPOSE_MIN_VERSION);
-
-        resolve(Process::class)
-            ->shouldReceive('dockerCompose')
-            ->withArgs(function ($command) {
-                return $command === 'ps';
-            })
-            ->andReturn(1);
-
-        $this->artisan('start --timeout=1')
-            ->expectsOutput('Timed out waiting the command to finish')
-            ->assertExitCode(1);
+        $this->mockChecker(
+            Checker::DOCKER_MIN_VERSION,
+            Checker::DOCKER_API_MIN_VERSION,
+            '0.0.0'
+        );
+
+        $this->artisan('start')->assertExitCode(1);
+    }
+
+    protected function mockChecker(
+        $dockerVersion,
+        $dockerApiVersion,
+        $dockerComposeVersion
+    ) {
+        $this->mock(Checker::class, function ($mock) use (
+            $dockerVersion,
+            $dockerApiVersion,
+            $dockerComposeVersion
+        ) {
+            $mock->shouldReceive('dockerVersion')
+                ->andReturn($dockerVersion);
+
+            $mock->shouldReceive('dockerApiVersion')
+                ->andReturn($dockerApiVersion);
+
+            $mock->shouldReceive('dockerComposeVersion')
+                ->andReturn($dockerComposeVersion);
+        })->makePartial();
     }
+
+    // public function testStartTimeout()
```

Comment:
is there something blocking this? may we remove it, or it is intended to be kept for future usage?

Reply:
shall we remove this?